README.md

```diff
@@ -9,103 +9,133 @@ This repository contains:
+If you need to re-generate the files in `./run` for any reason, you can simply delete it and it will be recreated the
+next time you launch a development build. This can help if you are running into errors related to the development
+environment setup.
 
-**Warning**: Make sure your locally-hosted development server & proxy are not visible from the Internet if you're launching in unsafe mode! Check your system (or gateway) firewall.
+### Additional setup (Linux users only)
 
-When debugging things, it's sometimes very useful to use breakpoints to analyze state and execute code line-by-line. Doing this on a Minecraft server works, but is tricky with a vanilla client due to the read timeout delay. As such, it is recommended to use a mod, like [this one](https://www.curseforge.com/minecraft/mc-mods/timeoutout-fabric), to increase this value to whatever you want. This will prevent your client from disconnecting itself due to a nonresponsive server.
+Many Linux users will not need to go through with this additional setup. To determine if it is necessary for you, run
+the `id` command. This will output some text like the following:
 
-### For production
+```
+uid=1000(steank) gid=1000(steank)
+```
 
-If the `unsafe` argument is not specified, the server *will refuse to run* with the default secret, for security reasons. You'll need to configure your own secret.
+Make note of your current user name (the one that owns all the project files; in this example it is `steank`) and
+its `uid` and `gid`. _If `uid` and `gid` are both 1000, you do not need to do anything else._ However, if one differs,
+you will need to add some additional configuration to `.override.env`:
 
-First, come up with an appropriate string. You can use a password manager, or a generation tool like `apg` or `pwgen`. If your server is accessible from the Internet, make sure it's suitably complex. Keep in mind that your secret isn't something you need to actually *remember*, and thus there's no real drawback to making it as long as you want.
+```
+# This example assumes your UID and GID are 1001
+PHANTAZM_UID='1001'
+PHANTAZM_GID='1001'
+```
 
-Next, set the `proxySecret` field in `./run/server-1/server-config.toml`, and the `forwarding-secret` field in `./run/velocity/velocity.toml`, to the same string (your secret).
+In other words, set `PHANTAZM_UID` to your `uid` and `PHANTAZM_GID` to your `gid`.
 
-**Warning**: The secret is stored, in plaintext, in both the server and proxy configuration files. This is, regrettably, something we can't do much about. Make sure access to your backend servers is properly secured.
+### Joining the local development server
 
-### Running
+Once you've set up your development environment and run `Launch Phantazm`, you should have a running proxy, server,
+and database. You can connect to the server (through the proxy) by joining `localhost` on a vanilla 1.19.4 Minecraft
+client.
```

Comment:
useful to clarify that it could be any version supported by via somehow if people are used to playing on 1.20.1 or something